src/smart-components/group/service-account/add-group-service-accounts.tsx

```diff
@@ -1,14 +1,171 @@
+import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import AppLink from '../../../presentational-components/shared/AppLink';
+import useAppNavigate from '../../../hooks/useAppNavigate';
+import { TableToolbarView } from '../../../presentational-components/shared/table-toolbar-view';
+import { fetchServiceAccounts } from '../../../redux/actions/service-account-actions';
+import { ServiceAccountsState } from '../../../redux/reducers/service-account-reducer';
+import { getDateFormat } from '../../../helpers/shared/helpers';
+import { RowProps } from '../../user/user-table-helpers';
+import { LAST_PAGE, ServiceAccount } from '../../../helpers/service-account/service-account-helper';
+import { DEFAULT_ACCESS_GROUP_ID } from '../../../utilities/constants';
+import { addServiceAccountsToGroup } from '../../../redux/actions/group-actions';
 import messages from '../../../Messages';
+import './group-service-accounts.scss';
 
-const AddGroupServiceAccounts: React.FC = () => {
+interface AddGroupServiceAccountsProps {
+  submitRoute: string;
+  cancelRoute: string;
+}
+
+export interface PaginationProps {
+  count?: number;
+  limit: number;
+  offset: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const reducer = ({ serviceAccountReducer, groupReducer: { systemGroup } }: { serviceAccountReducer: ServiceAccountsState; groupReducer: any }) => ({
+  serviceAccounts: serviceAccountReducer.serviceAccounts,
+  status: serviceAccountReducer.status,
+  isLoading: serviceAccountReducer.isLoading,
+  limit: serviceAccountReducer.limit,
+  offset: serviceAccountReducer.offset,
+  systemGroupUuid: systemGroup?.uuid,
+});
+
+const createRows = (data: ServiceAccount[], checkedRows = []) =>
+  data?.reduce(
+    (acc: unknown[], curr: ServiceAccount) => [
+      ...acc,
+      {
+        uuid: curr.uuid,
+        title: curr.name,
+        cells: [
+          curr.name,
+          curr.clientId,
+          curr.createdBy,
+          <Fragment key={`${curr.name}-modified`}>
+            <DateFormat date={curr.createdAt} type={getDateFormat(curr.createdAt)} />
+          </Fragment>,
+        ],
+        selected: Boolean(checkedRows && checkedRows.find((row: RowProps) => row.uuid === curr.uuid)),
+      },
+    ],
+    []
+  );
+
+const AddGroupServiceAccounts: React.FunctionComponent<AddGroupServiceAccountsProps> = ({
+  submitRoute,
+  cancelRoute,
+}: AddGroupServiceAccountsProps) => {
   const intl = useIntl();
+  const navigate = useAppNavigate();
+  const dispatch = useDispatch();
+  const { groupId } = useParams();
+  const { auth, getEnvironmentDetails } = useChrome();
+  const [selectedAccounts, setSelectedAccounts] = useState([]);
+  const { serviceAccounts, status, limit, offset, isLoading, systemGroupUuid } = useSelector(reducer);
+
+  const fetchAccounts = async (props?: PaginationProps) => {
```

Comment:
Can you wrap this into `useCallback`? So we don't re-create this function on each render.